public/chat_header_button.tsx

```diff
@@ -140,6 +142,26 @@ export const HeaderChatButton: React.FC<HeaderChatButtonProps> = (props) => {
     };
   }, []);
 
+  useEffect(() => {
```

Comment:
Shall we add some unit tests for these logic?

Reply:
definitely. wanted to get this out for feedback.